### PR TITLE
User form

### PR DESCRIPTION
**Pull request by nirdesh101**
- Made the functionality of adding user each time clicking the submit button

### CODE REVIEW

File: context-api/src/App.css [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+} 
+
+.login {
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  height: 90vh;
+  flex-direction: column;
+}
+
```
Comment (anisha2lc) [inline context-api/src/App.css:16]: lets use em instead of px

File: context-api/src/App.css [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  color: white;
+  border-radius: 5px;
+}
+
+button {
+  width: 8rem;
+  padding: 10px;
+  margin: 10px;
+  background-color:rgb(0, 168, 246);
+  border: 0px;
```
Comment (anisha2lc) [inline context-api/src/App.css:32]: same here

File: context-api/src/App.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
+import './App.css';
+import React, {useState} from 'react'
```
Comment (anisha2lc) [inline context-api/src/App.js:2]: lets put react import in the top and all the other imports below it

File: context-api/src/Components/Login.jsx [hunk 4]
```diff
@@ -0,0 +1,32 @@
+import React, { useState, useContext } from "react";
```
Comment (anisha2lc) [inline context-api/src/Components/Login.jsx:1]: lets add a single line spacing between react import and other imports

File: context-api/src/Components/Login.jsx [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+import { LoginContext } from "../Contexts/LoginContext";
+
+function Login() {
+
```
Comment (anisha2lc) [inline context-api/src/Components/Login.jsx:5]: lets remove unrequired line spacing

File: context-api/src/Components/Profile.jsx [hunk 6]
```diff
@@ -0,0 +1,16 @@
+import React, {useContext} from 'react'
+import { LoginContext } from "../Contexts/LoginContext";
+
+
```
Comment (anisha2lc) [inline context-api/src/Components/Profile.jsx:4]: lets add only one spacing in between line no. 2 and line no. 5

File: user-form/src/App.js [hunk 7]
```diff
@@ -0,0 +1,45 @@
+import React, { useState, useContext } from "react";
+import "./App.css";
```
Comment (anisha2lc) [inline user-form/src/App.js:3]: lets add spacing between react imports and other imports

File: user-form/src/App.js [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+  const [userData, setUserData] = useState(jsonData, "jsonData");
+
+  return (
+    <>
+      <Context>
+        <Nav />
+        <div className="container">
+          <Routes>
+            <Route
+              path="/"
```
Comment (anisha2lc) [inline user-form/src/App.js:37]: lets extract path in a variable and use in route.

const HOME={
INDEX: "/"
}


<Route path={HOME.INDEX} element={<Home userData={userData} setUserData={setUserData} />} />

File: user-form/src/App.js [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+            />
+            <Route
+              path="/userlist/:id"
+              element={
+                <Details userData={userData} setUserData={setUserData} />
+              }
+            />
+          </Routes>
+        </div>
+      </Context>
```
Comment (anisha2lc) [inline user-form/src/App.js]: lets use above approach for path in all other routes

File: user-form/src/Components/FormFill.jsx [hunk 10]
```diff
@@ -0,0 +1,230 @@
+import React, { useId, useState, useContext } from "react";
```
Comment (anisha2lc) [inline user-form/src/Components/FormFill.jsx:1]: lets add spacing between react and all the other imports

File: user-form/src/Components/FormFill.jsx [hunk 11]
```diff
@@ -0,0 +1,230 @@
...
+    if (localStorage.getItem("key")) {
+      setData(JSON.parse(localStorage.getItem("key")));
+    }
+  }, []);
+
+  useEffect(() => {
+    localStorage.setItem("key", JSON.stringify(data));
+  }, [data]);
+
+  console.log(data, "data");
```
Comment (anisha2lc) [inline user-form/src/Components/FormFill.jsx]: lets remove console from here

File: user-form/src/Components/Home.jsx [hunk 12]
```diff
@@ -0,0 +1,8 @@
+import React from "react";
```
Comment (anisha2lc) [inline user-form/src/Components/Home.jsx:1]: lets add spacing between react import and other imports

File: user-form/src/Components/NavBar.jsx [hunk 13]
```diff
@@ -0,0 +1,44 @@
...
+
+function Nav() {
+  return (
+    <nav className="navbar navbar-expand-lg navbar-light bg-dark bg-gradient">
+      <div className="container">
+        <h1>
+          <NavLink
+            className="navbar-brand fs-2 text-light fw-semibold me-5"
+            href="#"
+            to="/"
```
Comment (anisha2lc) [inline user-form/src/Components/NavBar.jsx:17]: we can use the same HOME.INDEX in Navlink.

File: user-form/src/Components/NavBar.jsx [hunk 14]
```diff
@@ -0,0 +1,44 @@
...
+          aria-controls="navbarNavAltMarkup"
+          aria-expanded="false"
+          aria-label="Toggle navigation"
+        >
+          <span className="navbar-toggler-icon"></span>
+        </button>
+        <div className="collapse navbar-collapse" id="navbarNavAltMarkup">
+          <div className="navbar-nav">
+            <NavLink
+              to="/userlist"
```
Comment (anisha2lc) [inline user-form/src/Components/NavBar.jsx]: same here

File: user-form/src/Components/UserList.jsx [hunk 15]
```diff
@@ -0,0 +1,88 @@
...
+  function init() {
+    const value = localStorage.getItem(key);
+
+    if (value) {
+      setUserData(JSON.parse(value));
+    }
+  }
+  useEffect(() => init(), []);
+
+  console.log(userData);
```
Comment (anisha2lc) [inline user-form/src/Components/UserList.jsx]: lets not forget to remove console while pushing your code

File: user-form/src/Components/UserList.jsx [hunk 16]
```diff
@@ -0,0 +1,88 @@
...
+
+  function handleDelete(index) {
+    const updatedData = [...userData];
+    updatedData.splice(index, 1);
+    setUserData(updatedData);
+    localStorage.setItem(key, JSON.stringify(updatedData));
+  }
+
```
Comment (anisha2lc) [inline user-form/src/Components/UserList.jsx]: lets only add single line spacing here

File: user-form/src/ContextApi/context.jsx [hunk 17]
```diff
@@ -0,0 +1,26 @@
+import React, { createContext, useState } from "react";
+import jsonData from "../data.json"
```
Comment (anisha2lc) [inline user-form/src/ContextApi/context.jsx:2]: spacing between react imports and other imports

File: context-api/src/App.css [hunk 18]
```diff
@@ -0,0 +1,35 @@
...
+  justify-content: center;
+  text-align: center;
+} 
+
+.login {
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  height: 90vh;
+  flex-direction: column;
```
Comment (anisha2lc) [inline context-api/src/App.css:14]: lets use bootstrap classes for these css properties.

File: context-api/src/App.css (hunk 2 above)
Comment (anisha2lc) [inline context-api/src/App.css:32]: lets use bootstrap classes wherever possible

File: context-api/src/App.css (hunk 18 above)
Comment (nirdesh101) [reply context-api/src/App.css:14]: i have not used bootstrap in this task

**Pull request merged**